datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -64,13 +71,28 @@ public class SdxHorizontalScalingService {
     @Inject
     private CloudbreakFlowService cloudbreakFlowService;
 
+    private static List<InstanceMetaDataV4Response> getGatewayMetaData(String gatewayInstanceName, Map<String, Set<InstanceMetaDataV4Response>> instanceMap) {
+        List<InstanceMetaDataV4Response> gatewayMetaData = instanceMap.get(gatewayInstanceName).stream()
+                .filter(data -> !data.getInstanceStatus().equals(InstanceStatus.SERVICES_HEALTHY)
+                        || !data.getInstanceStatus().equals(InstanceStatus.SERVICES_RUNNING))
+                .toList();
+        return gatewayMetaData;
+    }
+
```

Comment:
You are calling the method "getGatewayMetaData" but in fact you are ekipping metadata of the hosts that are not healthy or hosts that are not in SERVICES_RUNNING state. It is misleading.